packages/lib/raqb/resolveQueryValue.ts

```diff
@@ -0,0 +1,252 @@
+};
+
+function getFieldResponseValueAsLabel({
+  field,
+  fieldResponseValue,
+}: {
+  fieldResponseValue: dynamicFieldValueOperandsResponse[keyof dynamicFieldValueOperandsResponse]["value"];
+  field: {
+    type: string;
+    options?: {
+      id: string | null;
+      label: string;
+    }[];
+  };
+}) {
+  const nonNumberFieldResponseValue =
+    typeof fieldResponseValue === "number" ? fieldResponseValue.toString() : fieldResponseValue;
+
+  if (!field.options) {
+    return nonNumberFieldResponseValue;
+  }
+
+  if (nonNumberFieldResponseValue instanceof Array) {
+    return nonNumberFieldResponseValue.map(transformValue);
+  }
+
+  return transformValue(nonNumberFieldResponseValue);
+
+  function transformValue(idOrLabel: string) {
+    const foundOptionById = field.options?.find((option) => {
+      return option.id === idOrLabel;
+    });
+    if (foundOptionById) {
+      return foundOptionById.label;
+    } else {
+      return idOrLabel.toString();
+    }
+  }
+}
+
+/**
+ * Resolves field template placeholders ({field:fieldId}) in a JSON string with actual values from form responses.
+ * Handles arrays properly for JSONLogic compatibility by spreading array values when needed.
+ *
+ * @param queryValueString - JSON string containing RAQB query with field template placeholders like {field:fieldId}
+ * @param dynamicFieldValueOperands - Optional object containing fields metadata and user's form responses
+ * @param dynamicFieldValueOperands.fields - Array of field definitions with id, type, label, etc.
+ * @param dynamicFieldValueOperands.response - Object mapping field IDs to their response values
+ *
+ * @example
+ * ```ts
+ * resolveQueryValue({
+ *   queryValue: {
+ *   "id": "9b8ba888-0123-4456-b89a-b1984020d699",
+ *   "type": "group",
+ *   "children1": {
+ *       "a8a89bba-89ab-4cde-b012-31984020ebaa": {
+ *         "type": "rule",
+ *         "id": "a8a89bba-89ab-4cde-b012-31984020ebaa",
+ *         "properties": {
+ *             "field": "88be6104-d350-411d-aa0f-092b2deda257",
+ *             "operator": "multiselect_some_in",
+ *             "value": [
+ *               [
+ *                 "{field:0bf77a89-2bd0-4df7-9648-758014ba3189}",
+ *                 "899c846d-7c02-43dc-9057-c3f8c118d41f"
+ *               ]
+ *             ],
+ *             "valueSrc": [
+ *               "value"
+ *             ],
+ *             "valueError": [
+ *               null
+ *             ],
+ *             "valueType": [
+ *                 "multiselect"
+ *             ]
+ *         }
+ *       }
+ *     }
+ *   },
+ *   dynamicFieldValueOperands: {
+ *     fields: [
+ *       { id: "location", type: "MULTI_SELECT", label: "Location" }
+ *     ],
+ *     response: {
+ *       location: { value: ["Delhi", "Mumbai"] }
+ *     }
+ *   }
+ * })
+ * // Returns: '{"value": [["delhi", "mumbai"]], "operator": "multiselect_some_in"}'
+ * ```
+ */
+export const resolveQueryValue = ({
```

Comment:
This fn replaces the earlier _replaceFieldTemplateVariableWithOptionLabel_ which was working on queryValueString and it now works on queryValue as an object(which is the original form). 

It was required because string regex replacement can't handle all the scenarios.

Reply:
Also, this fn itself calls _replaceAttributeOptionIdsWithOptionLabel_ now, instead of letting the caller do it first before calling this fn.

This makes the job of this fn focussed on resolving the query value which involves.
1. Repacing optionId with optionLabel
2. Replacing {field:FIELD_ID} with the field's response